applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp

```diff
@@ -1492,6 +1493,23 @@ inline void TreeContactSearch<TDim, TNumNodes, TNumNodesMaster>::CreateAuxiliarC
     // Iterate in the conditions and create the new ones
     ConditionsArrayType& r_conditions_array = rContactModelPart.Conditions();
 
+    // In case of debug mode
+    if (mThisParameters["echo_level"].GetInt() > 0) {
```

Comment:
why don't you use `debug_mode` here too?
Then you wouldn't have to manipulate the echo lvl

Reply:
Done